tools/proofers/my_suggestions.php

```diff
@@ -556,7 +556,7 @@ function compare_array_by_priority($a, $b)
     return ($a["priority"] < $b["priority"]) ? -1 : 1;
 }
 
-function get_user_suggestion_criteria($username, $flush_cache = false)
+function get_user_suggestion_criteria(?string $username, $flush_cache = false)
```

Comment:
`$flush_cache` could be `bool` I think if you want to make this edit too:
```diff
diff --git a/tools/proofers/my_suggestions.php b/tools/proofers/my_suggestions.php
index 99667fe36..f5156540e 100644
--- a/tools/proofers/my_suggestions.php
+++ b/tools/proofers/my_suggestions.php
@@ -11,7 +11,7 @@ include_once($relPath.'graph_data.inc'); // get_round_backlog_stats()
 require_login();

 $verbose = get_integer_param($_GET, "verbose", 0, 0, 1);
-$flush_cache = get_integer_param($_GET, "flush_cache", 0, 0, 1);
+$flush_cache = get_integer_param($_GET, "flush_cache", 0, 0, 1) == true;

 $username = User::current_username();
 if (user_is_a_sitemanager() || user_is_proj_facilitator()) {
```

And looks like `$verbose` could get the same treatment, honestly. Not in this PR, but I think we probably need to create a `get_boolean_param()` function in `misc.inc` that returns true/false instead of faking it like we are doing now.

Reply:
While ^^ is true, on reflection I wonder if this change belongs in this PR which is focused on `gradual.inc`?

Reply:
And finally, I created an issue for adding a new boolean validation func in `misc.inc`: https://github.com/DistributedProofreaders/dproofreaders/issues/1254

Reply:
> While ^^ is true, on reflection I wonder if this change belongs in this PR which is focused on gradual.inc?

That is a fair question. I included this as this function is the only one that `get_pages_proofed_maybe_simulated` with `$username` and typing it ensured that PHPStan would catch a type mismatch.

I looked at the callers of the functions in `pinc/gradual.inc` and this one seemed like a simple change that increased our type safety. To be fair, I wasn't consistent as I didn't update [this call site](https://github.com/DistributedProofreaders/dproofreaders/blob/504652ea0bb4f4c4ede0fb2a34f7f7ffa1b2b490/tools/proofers/my_suggestions.php#L64) that would have required a much larger change. Happy to revert this if you prefer that we avoid those small, tangentially-related additions.

Reply:
Nah, what you have in the PR is good.